File: Examen_Corto/Octave/ejercicio12.m

function salida = calcular_y_guardar_areas()

    % Menú de selección
    fprintf('Seleccione la figura geométrica:\n');
    fprintf('1. Círculo\n');
    fprintf('2. Triángulo\n');
    fprintf('3. Cuadrado\n');
    fprintf('4. Rectángulo\n');

    opcion = input('Ingrese el número de la opción deseada: ');

    switch opcion
        case 1
            % Área de un círculo
            radio = input('Ingrese el radio del círculo: ');
            area = pi * radio^2;
            figura = 'Círculo';

        case 2
            % Área de un triángulo
            base = input('Ingrese la base del triángulo: ');
            altura = input('Ingrese la altura del triángulo: ');
            area = 0.5 * base * altura;
            figura = 'Triángulo';

        case 3
            % Área de un cuadrado
            lado = input('Ingrese el lado del cuadrado: ');
            area = lado^2;
            figura = 'Cuadrado';

        case 4
            % Área de un rectángulo
            base = input('Ingrese la base del rectángulo: ');
            altura = input('Ingrese la altura del rectángulo: ');
            area = base * altura;
            figura = 'Rectángulo';

        otherwise
            fprintf('Opción no válida.\n');
            salida = [];
            return;
    end

    % Mostrar el resultado
    fprintf('El área del %s es: %.4f\n', figura, area);

    % Crear la cadena de texto
    cadena_resultado = sprintf('Figura: %s, Área: %.4f', figura, area);

    % Guardar la cadena en la variable salida
    salida = struct('cadena_resultado', cadena_resultado);

    % Guardar el resultado en un archivo de texto
    guardar_en_archivo(cadena_resultado);

end

function guardar_en_archivo(cadena_resultado)
    % Abrir el archivo en modo de escritura
    archivo = fopen('ejercicio12.txt', 'a');

    % Verificar si se pudo abrir el archivo
    if archivo == -1
        fprintf('Error al abrir el archivo.\n');
        return;
    end

    % Escribir el resultado en el archivo
    fprintf(archivo, '%s\n', cadena_resultado);


    % Cerrar el archivo
    fclose(archivo);

        % Conexión a la base de datos PostgreSQL
        pkg load database % Cargar el paquete

        conn = pq_connect(setdbopts('dbname','ExamenCorto1','host','localhost',
            'port','5432','user','postgres','password','gabrielgrdb'));

        query = sprintf("INSERT INTO salida VALUES ('%s');", cadena_resultado);
        N = pq_exec_params(conn, query);


    fprintf('Resultado guardado en "ejercicio12.txt".\n');
end

File: Examen_Corto/Python/ejercicio12.py
import math
import psycopg2

def calcular_y_guardar_areas():
    # Menú de selección
    print('Seleccione la figura geométrica:')
    print('1. Círculo')
    print('2. Triángulo')
    print('3. Cuadrado')
    print('4. Rectángulo')

    opcion = int(input('Ingrese el número de la opción deseada: '))

    # Definir las variables de área y figura
    area = 0
    figura = ''

    if opcion == 1:
        # Área de un círculo
        radio = float(input('Ingrese el radio del círculo: '))
        area = math.pi * radio ** 2
        figura = 'Círculo'
    elif opcion == 2:
        # Área de un triángulo
        base = float(input('Ingrese la base del triángulo: '))
        altura = float(input('Ingrese la altura del triángulo: '))
        area = 0.5 * base * altura
        figura = 'Triángulo'
    elif opcion == 3:
        # Área de un cuadrado
        lado = float(input('Ingrese el lado del cuadrado: '))
        area = lado ** 2
        figura = 'Cuadrado'
    elif opcion == 4:
        # Área de un rectángulo
        base = float(input('Ingrese la base del rectángulo: '))
        altura = float(input('Ingrese la altura del rectángulo: '))
        area = base * altura
        figura = 'Rectángulo'
    else:
        print('Opción no válida.')
        return None

    # Mostrar el resultado
    print(f'El área del {figura} es: {area:.4f}')

    # Crear la cadena de texto
    cadena_resultado = f'Figura: {figura}, Área: {area:.4f}'

    # Guardar el resultado en un archivo de texto
    guardar_en_archivo(cadena_resultado)

def guardar_en_archivo(cadena_resultado):
    # Abrir el archivo en modo de escritura
    with open('ejercicio12.txt', 'a') as archivo:
        # Escribir el resultado en el archivo
        archivo.write(f'{cadena_resultado}\n')

    # Conexión a la base de datos PostgreSQL
    try:
        conn = psycopg2.connect(
            dbname='ExamenCorto1',
            host='localhost',
            port='5432',
            user='postgres',
            password='gabrielgrdb'
        )

        # Crear un cursor
        cursor = conn.cursor()

        # Ejecutar la consulta SQL
        query = f"INSERT INTO salida VALUES ('{cadena_resultado}')"
        cursor.execute(query)

        # Confirmar la transacción
        conn.commit()

        # Cerrar el cursor y la conexión
        cursor.close()
        conn.close()

        print('Resultado guardado en "resultados.txt" y en la base de datos.')
    except Exception as e:
        print(f'Error al guardar en la base de datos: {e}')

# Ejecutar la función principal
calcular_y_guardar_areas()
